engine/console/game.go

```diff
@@ -149,6 +149,43 @@ func (game *Game) handleGameEvent(event interface{}) {
 		}
 	case core.NewFood:
 		game.food = append(game.food, event.Pos)
+	case core.PlayerMove:
+		for id, dir := range event.Moves {
+			prevHead := game.snakes[id].head
+			// move snake's head
+			if dir == core.Up {
+				game.snakes[id].head = core.Coord{X: prevHead.X, Y: prevHead.Y - 1}
+			} else if dir == core.Left {
+				game.snakes[id].head = core.Coord{X: prevHead.X - 1, Y: prevHead.Y}
+			} else if dir == core.Right {
+				game.snakes[id].head = core.Coord{X: prevHead.X + 1, Y: prevHead.Y}
+			} else if dir == core.Down {
+				game.snakes[id].head = core.Coord{X: prevHead.X, Y: prevHead.Y + 1}
+			} else {
+				panic("the value of direction is unknown")
+			}
+			// move snakes body
+			if len(game.snakes[id].body) == 0 {
+				continue
+			}
+			for i := len(game.snakes[id].body) - 1; i > 0; i -= 1 {
+				game.snakes[id].body[i] = game.snakes[id].body[i - 1]
+			}
+			game.snakes[id].body[0] = prevHead
+		}
+	case core.FoodEaten:
+		idx := 0
+		for i, food := range game.food {
+			if food.X == event.Pos.X && food.Y == event.Pos.Y {
+				idx = i
+				break
+			}
+		}
+		game.food[idx] = game.food[len(game.food) - 1]
+		game.food[len(game.food) - 1] = core.Coord{X: 0, Y: 0}
```

Comment:
You can omit initializing X, Y fields, because they will be zero-initialized by default: `core.Coord{}`.